ansible/playbooks/aix.yml

```diff
@@ -6,125 +6,282 @@
+  ##########################################################################
+    - name: Transfer Openssl to remote host
       copy:
-        src: "{{ item }}"
-        dest: /tmp/playbook_tmp/rpms/
-        mode: 0755
-      with_fileglob:
-        - /home/ubuntu/aix/rpms/*
-      tags: additional_rpms
+        src: /Vendor_Files/aix/openssl-1.0.2.1300.tar.gz
+        dest: /tmp/openssl-1.0.2.1300.tar.gz
+      tags: openssl
+
+    - name: Extract Openssl 
+      shell: gzip -d /tmp/openssl-1.0.2.1300.tar.gz
+      tags: openssl
+    
+    - shell: tar -xvf /tmp/openssl-1.0.2.1300.tar -C /tmp/
+      tags: openssl
+
+    - name: Install IBM Openssl - installp
+      shell: installp -aXYgd /tmp/openssl-1.0.2.1300 openssl.base
+      register: result.openssl
+      ignore_errors: yes
+      tags: openssl
 
-    - name: Download script to install RPM packages
+    - name: Clean openssl tmp files
+      file:
+        path: "{{ item }}"
+        state: absent
+      with_items:
+        - /tmp/openssl-1.0.2.1300
+        - /tmp/openssl-1.0.2.1300.tar
+      tags: openssl
+
+  ####################################################
+  # Uninstall conflicting packages from base image   #
+  # if they were installed via rpm unless yum exists #
+  ####################################################
+    - stat:
+        path: /usr/bin/yum
+      register: yum
+
+    - debug:
+        msg: "Yum installed, skipping download and installation"
+      when: yum.stat.islnk is defined
+
+    - name: Uninstall conflicting packages
+      shell: rpm -e --nodeps $(rpm -qa | grep -E "cloud-init|perl|openssl") 2>/dev/null
+      when: yum.stat.islnk is not defined
+      tags: rpm_remove
+
+  ####################################
+  # Install yum and update to latest #    
+  ####################################
+    - name: Download yum.sh
       get_url:
-        url: https://raw.githubusercontent.com/AdoptOpenJDK/openjdk-infrastructure/master/ansible/playbooks/scripts/AIX_RPM_PKGs_Install.sh 
-        dest: /tmp/playbook_tmp/AIX_RPM_PKGs_Install.sh
+        url: ftp://public.dhe.ibm.com/aix/freeSoftware/aixtoolbox/ezinstall/ppc/yum.sh
+        validate_certs: False
+        dest: /tmp/
         mode: 0775
-        validate_certs: no
-      tags: additional_rpms
+        timeout: 25
+      when: yum.stat.islnk is not defined
+      tags: yum
 
-    - name: Execute script to install RPM packages
-      command: /tmp/playbook_tmp/AIX_RPM_PKGs_Install.sh
-      tags: additional_rpms
-
-    ##############
-    # IBM Java 8 #
-    ##############
-    - name: Transfer Java to remote host
-      copy:
-        src: /home/ubuntu/aix/java/Java8_64.jre.tar.gz
-        dest: /tmp/playbook_tmp/java/Java8_64.jre.tar.gz
+    - name: Install yum and dependencies
+      shell: /tmp/yum.sh
+      register: result.yum
+      ignore_errors: yes
+      when: yum.stat.islnk is not defined
+      tags: yum
+
+    - name: Ensure the yum package index is up to date
```

Comment:
This set of commands will update everything to latest not just updating the cache so, I think the title here should be changed